packages/helix-shared-storage/src/storage.js

```diff
@@ -312,8 +314,14 @@ class Bucket {
 
     try {
       if (opts.addMetadata) {
-        const meta = await this.metadata(key) ?? {};
-        input.Metadata = { ...meta, ...opts.addMetadata };
+        const headers = await this.head(key);
+        if (headers) {
+          input.ContentType = headers.ContentType;
+          input.ContentEncoding = headers.ContentEncoding;
+          input.CacheControl = headers.CacheControl;
+          input.ContentDisposition = headers.ContentDisposition;
+        }
```

Comment:
I think copying an object from a source that doesn't exist should simply fail here - I guess the `CopyObjectCommand` would do - and then you don't have to create tests that have `undefined` in `Content-Type` below.

Reply:
very good point 👍